exporter/stats/prometheus/prometheus.go

```diff
@@ -39,39 +47,86 @@ type Exporter struct {
 
 // Options contains options for configuring the exporter.
 type Options struct {
-	OnError func(err error)
+	Namespace string
+	OnError   func(err error)
 }
 
-type collector struct {
-	descs   []*prometheus.Desc
-	metrics []prometheus.Metric
+// NewExporter returns an exporter that exports stats to Prometheus.
+func NewExporter(o Options) (*Exporter, error) {
+	reg := prometheus.NewRegistry()
+	collector := newCollector(o, reg)
+	e := &Exporter{
+		opts:    o,
+		g:       reg,
+		c:       collector,
+		handler: promhttp.HandlerFor(reg, promhttp.HandlerOpts{}),
+	}
+	return e, nil
 }
 
-func (c *collector) Describe(ch chan<- *prometheus.Desc) {
-	panic("not implemented")
+var _ http.Handler = (*Exporter)(nil)
+var _ stats.Exporter = (*Exporter)(nil)
+var _ stats.ViewRegistrar = (*Exporter)(nil)
+
+// RegisterView is the hook that's run either intentionally
+// by the client or the stat's library will automatically invoke
+// it everytime that a new view is added.
+func (e *Exporter) RegisterView(view *stats.View) {
```

Comment:
Great question. The problem though is that we'd have to then run a new registration cycle for every view that comes in, which starts to get out hand since we have to check if the view was already registered, lest run `reg.Register(collector)` so that we'll have the `prometheus.Description`